From 47d874a96a2df144ad9459517e800dbd08f2d3e8 Mon Sep 17 00:00:00 2001
From: Richard Su <rwsu@redhat.com>
Date: Wed, 30 Apr 2014 15:26:31 -0700
Subject: [PATCH] Fix rabbitmq-server selinux labeling

/mnt/state/var/lib/rabbitmq should be labeled as rabbitmq_var_lib_t
and not var_lib_t.
/mnt/state/var/log/rabbitmq should be labeled as rabbitmq_var_log_t
and not var_log_t.

Create /mnt/state/var/log/rabbitmq directory in pre-configure.d
so that it can be labeled before rabbitmq-server is started.

Change-Id: Ifaee9755a58c514003e1c35e73f7b9a91854046f
---
 .../os-refresh-config/configure.d/20-rabbitmq-server-selinux   | 10 ++++++++++
 .../os-refresh-config/pre-configure.d/80-rabbitmq-cluster      |  1 +
 2 files changed, 11 insertions(+)
 create mode 100755 elements/rabbitmq-server/os-refresh-config/configure.d/20-rabbitmq-server-selinux

diff --git a/elements/rabbitmq-server/os-refresh-config/configure.d/20-rabbitmq-server-selinux b/elements/rabbitmq-server/os-refresh-config/configure.d/20-rabbitmq-server-selinux
new file mode 100755
index 0000000..3eea7f2
--- /dev/null
+++ b/elements/rabbitmq-server/os-refresh-config/configure.d/20-rabbitmq-server-selinux
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -eu
+
+[ -x /usr/sbin/semanage ] || exit 0
+
+semanage fcontext -a -t rabbitmq_var_lib_t "/mnt/state/var/lib/rabbitmq(/.*)?"
+restorecon -Rv /mnt/state/var/lib/rabbitmq
+
+semanage fcontext -a -t rabbitmq_var_log_t "/mnt/state/var/log/rabbitmq(/.*)?"
+restorecon -Rv /mnt/state/var/log/rabbitmq
diff --git a/elements/rabbitmq-server/os-refresh-config/pre-configure.d/80-rabbitmq-cluster b/elements/rabbitmq-server/os-refresh-config/pre-configure.d/80-rabbitmq-cluster
index 62b0cba..8dce997 100755
--- a/elements/rabbitmq-server/os-refresh-config/pre-configure.d/80-rabbitmq-cluster
+++ b/elements/rabbitmq-server/os-refresh-config/pre-configure.d/80-rabbitmq-cluster
@@ -3,3 +3,4 @@ set -eux
 
 [ -d /mnt/state/var/lib/rabbitmq ] || install -d -D -m 0770 -o rabbitmq -g rabbitmq /mnt/state/var/lib/rabbitmq
 install -m 600 -o rabbitmq -g rabbitmq /dev/null /mnt/state/var/lib/rabbitmq/.erlang.cookie
+[ -d /mnt/state/var/log/rabbitmq ] || install -d -D -m 0770 -o rabbitmq -g rabbitmq /mnt/state/var/log/rabbitmq
-- 
1.9.3

